staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/fairqueuing.go

```diff
@@ -0,0 +1,482 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fairqueuing
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service times
+// implementation of:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
+type FQScheduler struct {
+	lock                 sync.Mutex
+	Queues               []FQQueue
+	clock                clock.Clock
+	vt                   float64
+	estimatedServiceTime float64
+	lastRealTime         time.Time
+	robinidx             int
```

Comment:
We normally use lowerCamelCase, thus `robinIdx`.

Reply:
updated

Reply:
I still see lowercase.

Reply:
Sorry about this, I've pushed this now.  I will be more explicit in the future and push the commit with the change before posting that I've made the change.